keys.go

```diff
@@ -81,6 +81,18 @@ type KeysActionRequest struct {
 	Payload    string   `json:"payload,omitempty"`
 }
 
+//KeyVersion represents the key version information as returned by KP API
+type KeyVersion struct {
+	ID           string     `json:"id,omitempty"`
+	CreationDate *time.Time `json:"creationDate,omitempty"`
+}
+
+//KeyVersions represents the collection of all the versions of a Key
+type KeyVersions struct {
+	Metadata KeysMetadata `json:"metadata"`
+	Versions []KeyVersion `json:"resources"`
+}
```

Comment:
Lets make this a private struct and only use it for deserialization and return the []KeyVersions array. I don't think people will care about the metadata.

```suggestion
// keyVersions represents the collection of all the versions of a Key
type keyVersions struct {
	metadata KeysMetadata `json:"metadata"`
	versions []KeyVersion `json:"resources"`
}
```